pkg/stub/handler.go

```diff
@@ -104,6 +107,102 @@ type Handler struct {
+			return nil
+		}
+		cm = &corev1.ConfigMap{}
+		cm.Name = v1alpha1.SamplesResourceName
+		cm.Data = map[string]string{}
+		if len(srcfg.Spec.InstallType) == 0 {
+			cm.Data[installtypekey] = string(v1alpha1.CentosSamplesDistribution)
+		} else {
+			cm.Data[installtypekey] = string(srcfg.Spec.InstallType)
+		}
+		if len(srcfg.Spec.Architectures) == 0 {
+			cm.Data[x86] = x86
+		} else {
+			for _, arch := range srcfg.Spec.Architectures {
+				switch arch {
+				case x86:
+					cm.Data[x86] = x86
+				case ppc:
+					cm.Data[ppc] = ppc
+				}
+			}
+		}
+		_, err = h.configmapclientwrapper.Create(h.namespace, cm)
+		if err != nil {
+			// just return error to sdk for retry
+			return err
+		}
+	}
+	return nil
+}
+
+func (h *Handler) SpecValidation(srcfg *v1alpha1.SamplesResource) error {
+	// if we have not had a valid SamplesResource processed, allow caller to try with
+	// the srcfg contents
+	if h.samplesResource == nil || !h.samplesResource.ConditionTrue(v1alpha1.SamplesExist) {
+		return nil
+	}
+	cm, err := h.configmapclientwrapper.Get(h.namespace, v1alpha1.SamplesResourceName)
+	if cm == nil || err != nil {
+		// just return error to sdk for retry
+		return err
+	}
+
+	installtype, ok := cm.Data[installtypekey]
+	if !ok {
+		err = fmt.Errorf("could	not find the installtype in the config map %#v", cm.Data)
+		return h.processError(srcfg, v1alpha1.ConfigurationValid, corev1.ConditionUnknown, err, "%v")
+	}
+	switch srcfg.Spec.InstallType {
+	case "", v1alpha1.CentosSamplesDistribution:
+		if installtype != string(v1alpha1.CentosSamplesDistribution) {
+			err = fmt.Errorf("cannot change installtype from %s to %s", installtype, srcfg.Spec.InstallType)
+			return h.processError(srcfg, v1alpha1.ConfigurationValid, corev1.ConditionFalse, err, "%v")
+		}
+	case v1alpha1.RHELSamplesDistribution:
+		if installtype != string(v1alpha1.RHELSamplesDistribution) {
+			err = fmt.Errorf("cannot change installtype from %s to %s", installtype, srcfg.Spec.InstallType)
+			return h.processError(srcfg, v1alpha1.ConfigurationValid, corev1.ConditionFalse, err, "%v")
+		}
+	}
+
+	_, hasx86 := cm.Data[x86]
+	_, hasppc := cm.Data[ppc]
+
+	wantsx86 := false
+	wantsppc := false
+	if len(srcfg.Spec.Architectures) == 0 {
+		wantsx86 = true
+	}
+	for _, arch := range srcfg.Spec.Architectures {
+		switch arch {
+		case x86:
+			wantsx86 = true
+		case ppc:
+			wantsppc = true
+		}
+	}
+
+	if wantsx86 != hasx86 ||
+		wantsppc != hasppc {
+		err = fmt.Errorf("cannot change architectures from %#v to %#v", cm.Data, srcfg.Spec.Architectures)
+		return h.processError(srcfg, v1alpha1.ConfigurationValid, corev1.ConditionFalse, err, "%v")
+	}
+
+	return h.GoodConditionUpdate(srcfg, corev1.ConditionTrue, v1alpha1.ConfigurationValid)
+}
+
 func (h *Handler) GoodConditionUpdate(srcfg *v1alpha1.SamplesResource, newStatus corev1.ConditionStatus, condition v1alpha1.SamplesResourceConditionType) error {
 	status := srcfg.Condition(condition)
```

Comment:
nit but status should probably be named condition and condition should probably be named conditionType